### ISSUE TITLE
[CLI] Define dLLM Pydantic config model (unturtle_cli/config.py)

### ISSUE THREAD

**Issue by nishide-dev** [nishide-dev/unturtle#131]
## Background / Goal

`unsloth_cli/config.py` is designed for standard autoregressive LM training and lacks all dLLM-specific fields. This issue defines a new config hierarchy in `unturtle_cli/config.py` that covers masked diffusion training, generation, and GRPO.

## Acceptance Criteria

- `unturtle_cli/config.py` exports a root `Config` Pydantic model with the sections below
- `Config().model_dump()` round-trips through YAML without error
- `Config.load_config(path)` loads a YAML or JSON file
- `Config.apply_overrides(**kwargs)` merges CLI flag overrides into the config
- All dLLM-specific fields are present and documented with descriptions

### `ModelConfig`
| Field | Type | Default | Notes |
|-------|------|---------|-------|
| `model` | `Optional[str]` | `None` | HF model ID or local path |
| `model_type` | `Literal["auto","a2d","llada","dream"]` | `"auto"` | Hint to `FastDiffusionModel`; "auto" uses HF AutoConfig |
| `max_seq_length` | `int` | `2048` | |
| `load_in_4bit` | `bool` | `True` | |

### `DataConfig`
| Field | Type | Default |
|-------|------|---------|
| `dataset` | `Optional[str]` | `None` |
| `local_dataset` | `Optional[List[str]]` | `None` |
| `dataset_text_field` | `str` | `"text"` |

### `TrainingConfig`
| Field | Type | Default |
|-------|------|---------|
| `training_type` | `Literal["lora","full"]` | `"lora"` |
| `output_dir` | `str` | `"./outputs"` |
| `num_epochs` | `int` | `3` |
| `learning_rate` | `float` | `2e-4` |
| `batch_size` | `int` | `2` |
| `gradient_accumulation_steps` | `int` | `4` |
| `warmup_steps` | `int` | `5` |
| `max_steps` | `int` | `0` (disabled) |
| `save_steps` | `int` | `0` (disabled) |
| `weight_decay` | `float` | `0.01` |
| `random_seed` | `int` | `3407` |
| `packing` | `bool` | `False` |
| `gradient_checkpointing` | `Literal["unsloth","true","none"]` | `"unsloth"` |

### `DiffusionConfig` (dLLM-specific; maps to `DiffusionTrainingArguments`)
| Field | Type | Default |
|-------|------|---------|
| `alpha_scheduler` | `Literal["linear","cosine"]` | `"linear"` |
| `time_epsilon` | `float` | `1e-3` |
| `loss_weight_type` | `Literal["uniform","timestep","scheduler"]` | `"uniform"` |
| `completion_only` | `bool` | `True` |

### `LoraConfig`
| Field | Type | Default |
|-------|------|---------|
| `lora_r` | `int` | `64` |
| `lora_alpha` | `int` | `16` |
| `lora_dropout` | `float` | `0.0` |
| `target_modules` | `str` | `"q_proj,k_proj,v_proj,o_proj,gate_proj,up_proj,down_proj"` |
| `use_rslora` | `bool` | `False` |

### `LoggingConfig`
| Field | Type | Default |
|-------|------|---------|
| `enable_wandb` | `bool` | `False` |
| `wandb_project` | `str` | `"unturtle-training"` |
| `wandb_token` | `Optional[str]` | `None` (also reads `WANDB_API_KEY`) |
| `enable_tensorboard` | `bool` | `False` |
| `hf_token` | `Optional[str]` | `None` (also reads `HF_TOKEN`) |

### `GRPOConfig` (reserved for future `train --mode grpo`; fields optional for now)
| Field | Type | Default |
|-------|------|---------|
| `diffusion_steps` | `int` | `128` |
| `block_length` | `int` | `32` |
| `mask_id` | `Optional[int]` | `None` |
| `cfg_scale` | `float` | `0.0` |
| `remasking` | `Literal["random","low_confidence"]` | `"random"` |

### Root `Config`
```python
class Config(BaseModel):
    model: ModelConfig = ModelConfig()
    data: DataConfig = DataConfig()
    training: TrainingConfig = TrainingConfig()
    diffusion: DiffusionConfig = DiffusionConfig()
    lora: LoraConfig = LoraConfig()
    logging: LoggingConfig = LoggingConfig()
    grpo: Optional[GRPOConfig] = None
```

## Dependencies

- #129 (scaffold)

## Links

- Reference: `unsloth_cli/config.py`
- unturtle APIs: `unturtle/diffusion/trainer.py` (`DiffusionTrainingArguments`), `unturtle/diffusion/grpo_trainer.py` (`DiffuGRPOConfig`)
- Plan: `~/.claude/plans/hashed-tumbling-token.md` Issue C

**Comment by nishide-dev** [comment]
Closed by merged PR #143.
